### PR TITLE
Add build and test jobs for devconsole-operator

### PR DESCRIPTION
**Pull request by sbose78**
replaces https://github.com/openshift/release/pull/3259

- adds a CI step for "make build"
- adds a CI step for "make test"

In future, we might use intermediate builds to optimise this.

### CODE REVIEW

File: ci-operator/config/redhat-developer/devconsole-operator/redhat-developer-devconsole-operator-master.yaml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+resources:
+  '*':
+    limits:
+      memory: 1Gi
+    requests:
+      memory: 250Mi
+
+build_root:
+  project_image:
+    dockerfile_path: openshift-ci/build-root/Dockerfile
```
Comment (petr-muller) [inline ci-operator/config/redhat-developer/devconsole-operator/redhat-developer-devconsole-operator-master.yaml]: This is the reason why the `unit` rehearsal fails, this file does not seem to exist in your repo (https://github.com/redhat-developer/devconsole-operator). If you intend to add it there, it is fine to merge this PR as it is, rehearsals are not blocking.

File: ci-operator/config/redhat-developer/devconsole-operator/redhat-developer-devconsole-operator-master.yaml [hunk 2]
```diff
@@ -0,0 +1,37 @@
+canonical_go_repository: github.com/redhat-developer/devconsole-operator
+
+resources:
+  '*':
+    limits:
+      memory: 1Gi
+    requests:
+      memory: 250Mi
```
Comment (petr-muller) [inline ci-operator/config/redhat-developer/devconsole-operator/redhat-developer-devconsole-operator-master.yaml]: I think you'll want to add a CPU request too (but not limits).

File: ci-operator/config/redhat-developer/devconsole-operator/redhat-developer-devconsole-operator-master.yaml (hunk 1 above)
Comment (sbose78) [reply ci-operator/config/redhat-developer/devconsole-operator/redhat-developer-devconsole-operator-master.yaml]: Thanks! I've added the right path.

**Comment by sbose78** [comment]
@petr-muller @droslean This is ready for merge.

**Pull request merged**